### PR TITLE
13-implement data validation on home page

### PR DESCRIPTION
**Pull request by haroon-ali-dev**
#13

### CODE REVIEW

File: client/src/components/UserForm.js [hunk 1]
```diff
@@ -1,120 +1,181 @@
...
+import Button from 'react-bootstrap/Button';
 
 import "./UserForm.css";
 
+const schema = yup.object({
+  name: yup.string().min(3).max(50).matches(/^[a-zA-Z\s]+$/, 'must be letters only.').required(),
+
+  gender: yup.string().required(),
+
+  dateOfBirth: yup.string().required("Please enter a date.").matches(/^\d{4}-(0[1-9]|1[012])-(0[1-9]|[12][0-9]|3[01])$/, 'must be a valid date in the format DD/MM/YYYY'),
```
Comment (EneVladMihai) [inline client/src/components/UserForm.js:15]: There is a `.date()` format in yup which could simplify a bit this line and the `dateOfRelease` [see this resource](https://www.codedaily.io/tutorials/Yup-Date-Validation-with-Custom-Transform)

Comment (haroon-ali-dev) [reply client/src/components/UserForm.js]: done

File: client/src/components/UserForm.css [hunk 2]
```diff
@@ -9,6 +9,10 @@
   margin-bottom: 25px;
 }
 
+.container-input {
+  width: 50%;
```
Comment (EneVladMihai) [inline client/src/components/UserForm.css:13]: There are some utility classes in bootstrap. I'd recommend using them wherever possible as they are more verbose and you can avoid writing / maintaining CSS.
Here the utility class is called `w-50`.
https://getbootstrap.com/docs/5.0/utilities/sizing/

Comment (EneVladMihai) [reply client/src/components/UserForm.css:13]: Hmm, I see you were using this before. What motivated the change?

File: client/src/components/UserForm.css [hunk 3]
```diff
@@ -9,6 +9,10 @@
...
+}
+
 .container-btn {
   text-align: center;
```
Comment (EneVladMihai) [inline client/src/components/UserForm.css:17]: Boostrap class `text-center`

File: client/src/components/UserForm.css [hunk 4]
```diff
@@ -9,6 +9,10 @@
...
   margin-bottom: 50px;
```
Comment (EneVladMihai) [inline client/src/components/UserForm.css:18]: Bootsrap class `mb-[0-5]`. Better to use one of their predefiened values to ensure consistency throughout the app.
https://getbootstrap.com/docs/4.0/utilities/spacing/

**Review by EneVladMihai — CHANGES_REQUESTED** [review]
Have a look at bootstrap utility classes, whould they be helpful and reduce the need to write custom CSS?

File: client/src/components/UserForm.css (hunk 2 above)
Comment (haroon-ali-dev) [reply client/src/components/UserForm.css]: I didn't think about using it this way. I've made the change.

File: client/src/components/UserForm.css (hunk 3 above)
Comment (haroon-ali-dev) [reply client/src/components/UserForm.css]: I've made the change.

File: client/src/components/UserForm.css (hunk 4 above)
Comment (haroon-ali-dev) [reply client/src/components/UserForm.css]: I'm using that now.

**Review by EneVladMihai — APPROVED** [review]

**Pull request merged**